iothub/service/src/Common/Data/SharedAccessSignatureAuthorizationRule.cs

```diff
@@ -1,18 +1,25 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
 using System;
 using Newtonsoft.Json;
 
 namespace Microsoft.Azure.Devices.Common.Data
 {
-    public sealed class SharedAccessSignatureAuthorizationRule : IEquatable<SharedAccessSignatureAuthorizationRule>
+    /// <summary>
+    /// A shared access signature based authorization rule for authenticating requests against an IoT Hub.
+    /// </summary>
+    internal sealed class SharedAccessSignatureAuthorizationRule : IEquatable<SharedAccessSignatureAuthorizationRule>
     {
         private string _primaryKey;
         private string _secondaryKey;
 
         [JsonProperty(PropertyName = "keyName")]
```

Comment:
I cannot figure out what this field is supposed to signify. KeyName usually represents the shared access policy name but we already get that info via the `Rights` property

Reply:
Also, I have no idea why these properties have json annotations -> this object isn't something we'd generally serialialize or deserialize.

Reply:
Interesting, best to check with service team.